pkg/database/repository.go

```diff
@@ -57,31 +66,38 @@ func (r repository) Lock(id, instanceId, userId uint) (*model.Lock, error) {
 			Preload("Lock").
 			First(&d, id).Error
 		if err != nil {
+			if errors.Is(err, gorm.ErrRecordNotFound) {
+				err = errdef.NewNotFound("database not found by id: %d", id)
+			}
 			return err
 		}
 
 		if d.Lock != nil && d.Lock.InstanceID != 0 {
-			return fmt.Errorf("database already locked by user \"%d\" and instance \"%d\"", userId, d.Lock.InstanceID)
+			return errdef.NewBadRequest("database already locked by user %q and instance %q", userId, d.Lock.InstanceID)
 		}
 
 		lock = &model.Lock{
 			DatabaseID: id,
 			InstanceID: instanceId,
 			UserID:     userId,
 		}
-		err = tx.Create(lock).Error
-		if err != nil {
-			return err
-		}
-
-		return nil
+		return tx.Create(lock).Error
 	})
 
 	return lock, errTx
 }
 
 func (r repository) Unlock(id uint) error {
-	return r.db.Delete(&model.Lock{}, id).Error
+	db := r.db.Delete(&model.Lock{}, id)
+	if db.Error != nil {
+		return db.Error
+	}
+
+	if db.RowsAffected < 1 {
```

Comment:
@radnov made `Delete` return `NewNotFound` when deleting a user that does not exist. We should be consistent. If we want that behavior here as well we should move that logic into repository.Delete and call that method from here 😄

Reply:
But this is about deleting a lock, not a user?

Reply:
my bad I probably mistaken this as the service 🤦🏻 

should we use `r.db.Unscoped().Delete` as we mostly seem to do. the `r.db.Delete(&model.Lock{}, id)` does a soft-delete AFAIK

https://github.com/dhis2-sre/im-manager/pull/257/files#diff-c7bd001c6323abf762d3175df7f3e7c28aaa063386901005ee236254b39b007dR80

Reply:
Already pushed, I'm not sure why it's not reflected here :man_shrugging: